lib/linked_list.rb

```diff
@@ -19,43 +19,97 @@ def initialize
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     def add_first(value)
-      raise NotImplementedError
+      current = @head
+      @head = Node.new(value, current)
     end
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     def search(value)
-      raise NotImplementedError
+      if @head.nil?
+        return false
+      else
+        current = @head
+      end
+
+      while current
+        if current.data == value
+          return true
+        else
+          current = current.next
+        end
+      end

```

Comment:
Just for readability I suggest returning false when you can't find it.

```suggestion
      end
      return false
```